generated/nidcpower/nidcpower/unit_tests/test_nidcpower.py

```diff
@@ -0,0 +1,434 @@
+            }
+        ),
+        (
+            # python_init_params
+            {
+                "frequency": 300.0,
+                "ideal_capacitance": 5.0
+            },
+            # expected_python_members
+            {
+                "frequency": 300.0,
+                "reference_value_type": nidcpower.LCRReferenceValueType.IDEAL_CAPACITANCE,
+                "reference_value": 5.0
+            },
+            # expected_ctype_members
+            {
+                "frequency": 300.0,
+                "reference_value_type": nidcpower.LCRReferenceValueType.IDEAL_CAPACITANCE.value,
+                "reference_value_a": 5.0,
+                "reference_value_b": 0.0
+            }
+        ),
+        (
+            # python_init_params
+            {
+                "frequency": 400.0,
+                "ideal_inductance": 6.0
+            },
+            # expected_python_members
+            {
+                "frequency": 400.0,
+                "reference_value_type": nidcpower.LCRReferenceValueType.IDEAL_INDUCTANCE,
+                "reference_value": 6.0
+            },
+            # expected_ctype_members
+            {
+                "frequency": 400.0,
+                "reference_value_type": nidcpower.LCRReferenceValueType.IDEAL_INDUCTANCE.value,
+                "reference_value_a": 6.0,
+                "reference_value_b": 0.0
+            }
+        ),
+        (
+            # python_init_params
+            {
+                "frequency": 500.0,
+                "ideal_resistance": 7.0
+            },
+            # expected_python_members
+            {
+                "frequency": 500.0,
+                "reference_value_type": nidcpower.LCRReferenceValueType.IDEAL_RESISTANCE,
+                "reference_value": 7.0
+            },
+            # expected_ctype_members
+            {
+                "frequency": 500.0,
+                "reference_value_type": nidcpower.LCRReferenceValueType.IDEAL_RESISTANCE.value,
+                "reference_value_a": 7.0,
+                "reference_value_b": 0.0
+            },
+        ),
+    ],
+)
+def test_lcr_load_compensation_spot(
+    python_init_params,
+    expected_python_members,
+    expected_ctype_members
+):
+    python_object = nidcpower.LCRLoadCompensationSpot(**python_init_params)
+    for member in expected_python_members:
+        assert getattr(python_object, member) == pytest.approx(
+            expected_python_members[member]
+        )
+
+    ctype_object = nidcpower.struct_NILCRLoadCompensationSpot(python_object)
+    for member in expected_ctype_members:
+        assert getattr(ctype_object, member) == pytest.approx(
+            expected_ctype_members[member]
+        )
+
+
+def test_lcr_load_compensation_spot_repr_and_str():
+    python_object = nidcpower.LCRLoadCompensationSpot(frequency=100.0, impedance=complex(1.0, 2.0))
+    assert repr(python_object) == "LCRLoadCompensationSpot(frequency=100.0, impedance=(1+2j))"
```

Comment:
You should be able to create a valid python object using the output of `repr`, by calling `eval` on it. Verify this is doable by something like:
```
recreated_object = eval(repr(python_object))
assert str(recreated_object) == str(python_object)
```

Reply:
Thanks for the suggestion, I have this test to `test_lcr_load_compensation_spot_repr_and_str()` in my latest commit.

Note that this cannot be accomplished for LCRMeasurement which can now only be constructed from an instance of  `struct_NILCRMeasurement`, hence I have removed the custom `__repr__()` method for `LCRMeasurement`.